model/src/test/scala/org/broadinstitute/dsde/rawls/model/DataReferenceModelSpec.scala

```diff
@@ -55,6 +55,135 @@ class DataReferenceModelSpec extends AnyFreeSpec with Matchers {
+                   "metadata": {
+                     "cloningInstructions":"COPY_NOTHING",
+                     "description":"im a lil snapshot",
+                     "name":"snapshot1",
+                     "resourceId":"$snapshotResourceId",
+                     "resourceType":"DATA_REPO_SNAPSHOT",
+                     "stewardshipType":"REFERENCED",
+                     "workspaceId":"$workspaceId"
+                   },
+                   "resourceAttributes": {
+                     "gcpDataRepoSnapshot": {
+                        "instanceName":"test-instance",
+                        "snapshot":"$snapshotId"
+                     }
+                   }
+                 }
+               ]
+             }
+             """.parseJson
+        } {
+          new ResourceList().resources(
+            List(
+              new ResourceDescription()
+                .metadata(
+                  new ResourceMetadata()
+                    .name("snapshot1")
+                    .description("im a lil snapshot")
+                    .resourceId(snapshotResourceId)
+                    .resourceType(ResourceType.DATA_REPO_SNAPSHOT)
+                    .stewardshipType(StewardshipType.REFERENCED)
+                    .workspaceId(workspaceId)
+                    .cloningInstructions(CloningInstructionsEnum.NOTHING)
+
+                )
+                .resourceAttributes(
+                  new ResourceAttributesUnion()
+                    .gcpDataRepoSnapshot(
+                      new DataRepoSnapshotAttributes()
+                        .instanceName("test-instance")
+                        .snapshot(snapshotId.toString)
+                    ))
+            ).asJava
+          ).toJson
+        }
+      }
+
+      "Parsing ResourceMetadata should fail if resource id string is not a UUID" in {
+        val workspaceId = UUID.randomUUID()
+        assertThrows[DeserializationException] {
+          s"""
+             {
+               "cloningInstructions": "COPY_NOTHING",
+               "description": "im a lil snapshot",
+               "name": "snapshot1",
+               "resourceId": "not-an-id",
+               "resourceType": "DATA_REPO_SNAPSHOT",
+               "stewardshipType": "REFERENCED",
+               "workspaceId": "$workspaceId"
+              }""".parseJson.convertTo[ResourceMetadata]
+        }
+      }
+
+      "should fail if workspace id string is not a UUID" in {
+        val resourceId = UUID.randomUUID()
+        assertThrows[DeserializationException] {
+          s"""
+             {
+               "cloningInstructions":"COPY_NOTHING",
+               "description": "im a lil snapshot",
+               "name": "snapshot1",
+               "resourceId": "$resourceId",
+               "resourceType": "DATA_REPO_SNAPSHOT",
+               "stewardshipType": "REFERENCED",
+               "workspaceId": "not-an-id"
+              }""".parseJson.convertTo[ResourceMetadata]
+        }
+      }
+
```

Comment:
maybe add unit tests here that include deserializing nulls for various objects, to see how they behave